packages/config-transformer/tests/Converter/ConfigFormatConverter/YamlToPhp/Fixture/normal/import_already_converted_test.yaml

```diff
@@ -0,0 +1,12 @@
+imports:
+    - { resource: import_already_converted.yaml }
```

Comment:
this yaml file should not exist just the already convert php file then the `imports` fails when the delegatingLoader trys to load this file: https://github.com/symplify/symplify/blob/d1e786123d40523bc15e1e703e79ea1f457decf9/packages/config-transformer/src/ConfigLoader.php#L80

Reply:
Yes, at the moment the file does not exist, as nobody copied it there.

I think the test is ok now. It should pass without depending on file existance/non-existance.

Reply:
Is the `$delegatingLoader->load($fileRealPath);` required? or can we just remove it?

Reply:
It might be needed for some XML magic that loads nested configs to PHP, but I don't remember exactly.

What happens if you remove it?

Reply:
Lot of more errors happens:

<details><summary>Errors:</summary>

```bash
PHPUnit 9.5.21 #StandWithUkraine

Runtime:       PHP 8.1.8
Configuration: /Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/phpunit.xml

FFFFFF...........F....................................E.......... 65 / 75 ( 86%)
..........                                                        75 / 75 (100%)

Time: 00:02.086, Memory: 22.00 MB

There was 1 error:

1) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\YamlToPhp\YamlToPhpTest::testEcs with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
Error: Interface "PHP_CodeSniffer\Sniffs\Sniff" not found

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/YamlToPhp/Source/SomeSniff.php:10
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/ExprResolver/StringExprResolver.php:116
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/ExprResolver/StringExprResolver.php:60
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ArgsNodeFactory.php:107
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ArgsNodeFactory.php:75
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/CaseConverter/ConfiguredServiceCaseConverter.php:34
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ContainerConfiguratorReturnClosureFactory.php:213
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ContainerConfiguratorReturnClosureFactory.php:86
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ContainerConfiguratorReturnClosureFactory.php:64
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/vendor/symplify/php-config-printer/src/NodeFactory/ContainerConfiguratorReturnClosureFactory.php:47
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/src/Converter/YamlToPhpConverter.php:51
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/src/Converter/YamlToPhpConverter.php:39
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/src/Converter/ConfigFormatConverter.php:48
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:49
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/YamlToPhp/YamlToPhpTest.php:170
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/YamlToPhp/YamlToPhpTest.php:77

--

There were 7 failures:

1) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #0 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/single_service_tag.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $services = $containerConfigurator->services();
-
-    $services->set('mime_types')
-        ->tag('controller.service_subscriber')
-        ->tag('controller.service_arguments');
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

2) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #1 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/doctrine_config.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $containerConfigurator->extension('doctrine', ['dbal' => ['url' => '%env(resolve:DATABASE_URL)%']]);
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

3) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #2 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/multiple_tags.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $services = $containerConfigurator->services();
-
-    $services->set('mime_types')
-        ->tag('tag1')
-        ->tag('tag2');
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

4) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #3 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/tag_with_additional_attributes.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $services = $containerConfigurator->services();
-
-    $services->set('mime_types')
-        ->tag('tag1', ['priority' => 1])
-        ->tag('tag2', ['priority' => 2]);
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

5) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #4 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/no_import_inling.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $containerConfigurator->import(__DIR__ . '/some.php');
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

6) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\XmlToPhp\XmlToPhpTest::test with data set #5 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/XmlToPhp/Fixture/some.xml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@
-'<?php
-
-declare(strict_types=1);
-
-use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
-use Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\Source\MimeTypes;
-use function Symfony\Component\DependencyInjection\Loader\Configurator\service;
-
-return static function (ContainerConfigurator $containerConfigurator): void {
-    $services = $containerConfigurator->services();
-
-    $services->set('mime_types', MimeTypes::class)
-        ->args([''])
-        ->call('setDefault', [service('mime_types')])
-        ->call('setExtra', ['10000']);
-};
-'
+''

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:38
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/XmlToPhp/XmlToPhpTest.php:24

7) Symplify\ConfigTransformer\Tests\Converter\ConfigFormatConverter\YamlToPhp\YamlToPhpTest::testNormal with data set #5 (Symplify\SmartFileSystem\SmartFileInfo Object (...))
tests/Converter/ConfigFormatConverter/YamlToPhp/Fixture/normal/properties.yaml
Failed asserting that two strings are identical.
--- Expected
+++ Actual
@@ @@

 declare(strict_types=1);

-use PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff;
 use Symfony\Component\DependencyInjection\Loader\Configurator\ContainerConfigurator;
 use function Symfony\Component\DependencyInjection\Loader\Configurator\service;

@@ @@
 return static function (ContainerConfigurator $containerConfigurator): void {
     $services = $containerConfigurator->services();

-    $services->set(LineLengthSniff::class)
+    $services->set('PHP_CodeSniffer\Standards\Generic\Sniffs\Files\LineLengthSniff')
         ->property('absoluteLineLimit', 120);

     $services->set('app.newsletter_manager', 'App\Mail\NewsletterManager')

/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.php:52
/Users/alexanderschranz/Documents/Projects/symplify/packages/config-transformer/tests/Converter/ConfigFormatConverter/AbstractConfigFormatConverterTest.p
```

</details>

Reply:
catching the exception seems to work.